tsl/src/compression/compression.c

```diff
@@ -2265,8 +2265,13 @@ build_update_delete_scankeys(RowDecompressor *decompressor, List *filters, int *
  * This method will:
  *  1.scan compressed chunk
  *  2.decompress the row
- *  3.insert decompressed rows to uncompressed chunk
- *  4.delete this row from compressed chunk
+ *  3.delete this row from compressed chunk
+ *  4.insert decompressed rows to uncompressed chunk
+ *
+ * Return value:
+ * if all 4 steps defined above pass set chunk_status_changed to true and return true
```

Comment:
I think the following is true:
* the method will return true iff it processed all candidate rows
* `chunk_status_changed` will be true iff there are rows decompressed

note:
because it only return true or an error right now - I think `chunk_status_changed` is not necessary anymore...